osbenchmark/worker_coordinator/runner.py

```diff
@@ -1137,12 +1191,22 @@ def calculate_recall(predictions, neighbors, top_k):
                     continue
                 candidates.append(field_value)
             neighbors_dataset = params["neighbors"]
-            num_neighbors = params.get("k", 1)
-            recall_k = calculate_recall(candidates, neighbors_dataset, num_neighbors)
-            result.update({"recall@k": recall_k})
 
-            recall_1 = calculate_recall(candidates, neighbors_dataset, 1)
-            result.update({"recall@1": recall_1})
+            if params.get("k"):
```

Comment:
how about raise error if k and max/min is provided by params?

Reply:
i see validation added there. Can you explicitly check for params like k at line 1201 instead of else. We don't want catch all clause if it is not k.

Reply:
updated